sdv/metadata/single_table.py

```diff
@@ -22,6 +32,63 @@ class SingleTableMetadata:
     KEYS = ['columns', 'primary_key', 'alternate_keys', 'constraints', 'SCHEMA_VERSION']
     SCHEMA_VERSION = 'SINGLE_TABLE_V1'
 
+    @staticmethod
+    def _validate_numerical(column_name, **kwargs):
+        representations = [
+            'int', 'int64', 'int32', 'int16', 'int8',
+            'uint', 'uint64', 'uint32', 'uint16', 'uint8'
+            'float', 'float64', 'float32', 'float16', 'float8'
+        ]
+        representation = kwargs.get('representation')
+        if representation and representation not in representations:
+            raise ValueError(
+                f"Invalid value for 'representation' {representation} for column '{column_name}'.")
+
+    @staticmethod
+    def _validate_datetime(column_name, **kwargs):
+        datetime_format = kwargs.get('datetime_format')
+        if datetime_format:
+            formated_date = datetime.now().strftime(datetime_format)
+            matches = re.findall('(%.)|(%)', formated_date)
+            if matches:
+                matches = ', '.join(item for match in matches for item in match if item)
+                matches = ''.join(char if not char.isalnum() else '.' for char in matches)
```

Comment:
what is this regex doing? Can we just use the same logic as in the actual transformer? As in try to parse the datetime and if it comes back with an error than we catch that here?

Reply:
There is no such way to tell if a `strftime` is valid (or at least I didn't find one). This `regex` detects if there are `%` left on the `formated_date`, which is a way to tell whether or not the user passed an extra `%` which is the character that is usually used to specify any datetime like formating string.
The last line maybe I can remove it since it's just to clear the string a bit. I don't want ot confuse the end user since it can be a bit confusing if we find a `%` next to a `1` that comes from the `datetime.now()`.